============================= test session starts ==============================
platform darwin -- Python 3.8.3, pytest-6.1.2, py-1.9.0, pluggy-0.13.1 -- /Users/meganfrenkel/Workspace/KERMit/server/venv/bin/python
cachedir: .pytest_cache
rootdir: /Users/meganfrenkel/Workspace/KERMit
plugins: mock-3.3.1
collecting ... collected 132 items

server/tests/integration/test_authentication.py::test_registration_missing_header PASSED [  0%]
server/tests/integration/test_authentication.py::test_registration_malformed_header PASSED [  1%]
server/tests/integration/test_authentication.py::test_registration_verification_token_invalid PASSED [  2%]
server/tests/integration/test_authentication.py::test_registration_user_info_error PASSED [  3%]
server/tests/integration/test_authentication.py::test_register_token_valid_missing_info1 PASSED [  3%]
server/tests/integration/test_authentication.py::test_register_token_valid_missing_info2 PASSED [  4%]
server/tests/integration/test_authentication.py::test_registration_user_id_already_exists PASSED [  5%]
server/tests/integration/test_authentication.py::test_successful_user_registration PASSED [  6%]
server/tests/integration/test_leaderboard.py::test_get_leaders_no_completed_puzzle PASSED [  6%]
server/tests/integration/test_leaderboard.py::test_get_leaders_completed_puzzles PASSED [  7%]
server/tests/integration/test_leaderboard.py::test_get_leaders_completed_puzzles_limit2 PASSED [  8%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_use_game_without_registration PASSED [  9%]
server/tests/integration/test_sudoku_endpoints.py::test_get_all_puzzles_no_puzzles PASSED [  9%]
server/tests/integration/test_sudoku_endpoints.py::test_get_nonexistent_puzzle PASSED [ 10%]
server/tests/integration/test_sudoku_endpoints.py::test_save_new_puzzle_valid PASSED [ 11%]
server/tests/integration/test_sudoku_endpoints.py::test_save_new_puzzle_invalid_difficulty_small PASSED [ 12%]
server/tests/integration/test_sudoku_endpoints.py::test_save_new_puzzle_invalid_difficulty_large PASSED [ 12%]
server/tests/integration/test_sudoku_endpoints.py::test_save_new_puzzle_invalid_size_large PASSED [ 13%]
server/tests/integration/test_sudoku_endpoints.py::test_save_new_puzzle_invalid_size_small PASSED [ 14%]
server/tests/integration/test_sudoku_endpoints.py::test_get_all_puzzles_for_user PASSED [ 15%]
server/tests/integration/test_sudoku_endpoints.py::test_get_puzzle_valid PASSED [ 15%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_get_unaffiliated_puzzle PASSED [ 16%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_add_player_to_puzzle_already_in_puzzle PASSED [ 17%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_add_player_to_puzzle_that_doesnt_exist PASSED [ 18%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_add_player_to_puzzle_valid PASSED [ 18%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_to_join_puzzle_max_players_reached PASSED [ 19%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_valid_no_value_yet PASSED [ 20%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_valid_override_value PASSED [ 21%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_player_is_not_affiliated_with PASSED [ 21%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_invalid_piece_low PASSED [ 22%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_invalid_piece_high PASSED [ 23%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_invalid_position_high PASSED [ 24%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_add_piece_invalid_position_low PASSED [ 25%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_remove_piece PASSED [ 25%]
server/tests/integration/test_sudoku_endpoints.py::test_attempt_remove_static_piece PASSED [ 26%]
server/tests/integration/test_sudoku_endpoints.py::test_get_puzzle_solution_incomplete PASSED [ 27%]
server/tests/unit/test_authentication.py::test_authorize_token_missing_header PASSED [ 28%]
server/tests/unit/test_authentication.py::test_authorize_token_missing_header2 PASSED [ 28%]
server/tests/unit/test_authentication.py::test_authorize_token_malformed_header PASSED [ 29%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_error PASSED [ 30%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_success_register PASSED [ 31%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_success PASSED [ 31%]
server/tests/unit/test_authentication.py::test_authorize_token_validation_not_registered PASSED [ 32%]
server/tests/unit/test_authentication.py::test_register PASSED           [ 33%]
server/tests/unit/test_authentication.py::test_register_missing_info_email PASSED [ 34%]
server/tests/unit/test_authentication.py::test_register_missing_info_id PASSED [ 34%]
server/tests/unit/test_authentication.py::test_register_error_googleauth PASSED [ 35%]
server/tests/unit/test_authentication.py::test_register_exception PASSED [ 36%]
server/tests/unit/test_authentication.py::test_register_no_user_yet PASSED [ 37%]
server/tests/unit/test_authentication.py::test_register_exception_db PASSED [ 37%]
server/tests/unit/test_google_auth.py::test_validate_token PASSED        [ 38%]
server/tests/unit/test_google_auth.py::test_get_user_information PASSED  [ 39%]
server/tests/unit/test_leaderboard.py::test_get_leaderboard_no_leaders PASSED [ 40%]
server/tests/unit/test_leaderboard.py::test_get_leaderboard PASSED       [ 40%]
server/tests/unit/test_player.py::test_save_commit PASSED                [ 41%]
server/tests/unit/test_player.py::test_save PASSED                       [ 42%]
server/tests/unit/test_player.py::test_find_all_puzzles_for_player PASSED [ 43%]
server/tests/unit/test_player.py::test_find_players_for_puzzle PASSED    [ 43%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_already_too_many_players PASSED [ 44%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_player_doesnt_exist PASSED [ 45%]
server/tests/unit/test_player.py::test_add_player_to_puzzle_ok PASSED    [ 46%]
server/tests/unit/test_player.py::test_to_player_to_str PASSED           [ 46%]
server/tests/unit/test_puzzle_exception.py::test_get_message PASSED      [ 47%]
server/tests/unit/test_puzzle_pieces.py::test_find_all_pieces PASSED     [ 48%]
server/tests/unit/test_puzzle_pieces.py::test_get_piece PASSED           [ 49%]
server/tests/unit/test_puzzle_pieces.py::test_get_piece_none PASSED      [ 50%]
server/tests/unit/test_puzzle_pieces.py::test_save_autocommit PASSED     [ 50%]
server/tests/unit/test_puzzle_pieces.py::test_save PASSED                [ 51%]
server/tests/unit/test_puzzle_pieces.py::test_update_success PASSED      [ 52%]
server/tests/unit/test_puzzle_pieces.py::test_update_fail_static PASSED  [ 53%]
server/tests/unit/test_puzzle_pieces.py::test_get_as_str PASSED          [ 53%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_valid_defaults PASSED [ 54%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_valid_specification PASSED [ 55%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_str PASSED [ 56%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_too_low PASSED [ 56%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_difficulty_too_high PASSED [ 57%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_str PASSED [ 58%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_too_low PASSED [ 59%]
server/tests/unit/test_sudoku_puzzle.py::test_create_sudoku_puzzle_invalid_size_too_high PASSED [ 59%]
server/tests/unit/test_sudoku_puzzle.py::test_set_point_value1 PASSED    [ 60%]
server/tests/unit/test_sudoku_puzzle.py::test_set_point_value2 PASSED    [ 61%]
server/tests/unit/test_sudoku_puzzle.py::test_get_pieces_as_arr_all PASSED [ 62%]
server/tests/unit/test_sudoku_puzzle.py::test_get_pieces_as_arr_static_only PASSED [ 62%]
server/tests/unit/test_sudoku_puzzle.py::test_recreate_original_puzzle PASSED [ 63%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_null_values PASSED [ 64%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_complete PASSED [ 65%]
server/tests/unit/test_sudoku_puzzle.py::test_check_for_completion_incorrect_values PASSED [ 65%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_none PASSED [ 66%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_many_incorrect PASSED [ 67%]
server/tests/unit/test_sudoku_puzzle.py::test_check_discrepancies_incomplete_puzzle PASSED [ 68%]
server/tests/unit/test_sudoku_puzzle.py::test_get_puzzle_none PASSED     [ 68%]
server/tests/unit/test_sudoku_puzzle.py::test_get_puzzle_found PASSED    [ 69%]
server/tests/unit/test_sudoku_puzzle.py::test_save PASSED                [ 70%]
server/tests/unit/test_sudoku_puzzle.py::test_save_autocommit PASSED     [ 71%]
server/tests/unit/test_sudoku_puzzle.py::test_attempt_update_complete_puzzle PASSED [ 71%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate1 PASSED [ 72%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate2 PASSED [ 73%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate3 PASSED [ 74%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_coordinate4 PASSED [ 75%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_value1 PASSED [ 75%]
server/tests/unit/test_sudoku_puzzle.py::test_update_invalid_value2 PASSED [ 76%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid PASSED        [ 77%]
server/tests/unit/test_sudoku_puzzle.py::test_update_valid_complete_puzzle PASSED [ 78%]
server/tests/unit/test_sudoku_puzzle.py::test_set_puzzle_complete PASSED [ 78%]
server/tests/unit/test_sudoku_puzzle.py::test_as_str PASSED              [ 79%]
server/tests/unit/test_sudoku_resources.py::test_sudoku_to_json PASSED   [ 80%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_none PASSED [ 81%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_all PASSED [ 81%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_create_one_known_exception PASSED [ 82%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_create_one_unknown_exception PASSED [ 83%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_create_one PASSED [ 84%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_none_retrieved PASSED [ 84%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_none_associated PASSED [ 85%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzle_found PASSED [ 86%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_already_joined PASSED [ 87%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle PASSED [ 87%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_known_exception PASSED [ 88%]
server/tests/unit/test_sudoku_resources.py::test_join_sudoku_puzzle_unknown_exception PASSED [ 89%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_move_not_associated PASSED [ 90%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_move PASSED [ 90%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_move_invalid PASSED [ 91%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_add_move_exception PASSED [ 92%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_delete_piece PASSED [ 93%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_delete_piece_invalid PASSED [ 93%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_get_solution_not_associated PASSED [ 94%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_get_solution_not_associated_2 PASSED [ 95%]
server/tests/unit/test_sudoku_resources.py::test_get_sudoku_puzzles_get_solution PASSED [ 96%]
server/tests/unit/test_user.py::test_create_user PASSED                  [ 96%]
server/tests/unit/test_user.py::test_user_as_str PASSED                  [ 97%]
server/tests/unit/test_user.py::test_user_print PASSED                   [ 98%]
server/tests/unit/test_user.py::test_user_save PASSED                    [ 99%]
server/tests/unit/test_user.py::test_find_by_g_id PASSED                 [100%]

=============================== warnings summary ===============================
server/venv/lib/python3.8/site-packages/eventlet/patcher.py:1
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/eventlet/patcher.py:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

server/venv/lib/python3.8/site-packages/dns/hash.py:23
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/dns/hash.py:23: DeprecationWarning: dns.hash module will be removed in future versions. Please use hashlib instead.
    warnings.warn(

server/venv/lib/python3.8/site-packages/dns/namedict.py:35
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/dns/namedict.py:35: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    class NameDict(collections.MutableMapping):

server/tests/integration/test_authentication.py::test_registration_missing_header
  /Users/meganfrenkel/Workspace/KERMit/server/venv/lib/python3.8/site-packages/sqlalchemy/util/langhelpers.py:254: SADeprecationWarning: The 'postgres' dialect name has been renamed to 'postgresql'
    loader = self.auto_fn(name)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
======================= 132 passed, 4 warnings in 50.60s =======================
